android-json-form-wizard/src/main/java/com/vijay/jsonwizard/utils/Utils.java

```diff
@@ -214,6 +214,19 @@ public static void postEvent(BaseEvent event) {
         EventBus.getDefault().post(event);
     }
 
+    public static JSONObject getFieldFromJsonArray(String key, JSONArray jsonArray) {
+        JSONObject jsonObject = new JSONObject();
+        for (int i = 0; i < jsonArray.length(); i++) {
+            jsonObject = jsonArray.optJSONObject(i);
+            if (jsonObject != null) {
+                if (jsonObject.has(key)) {
```

Comment:
Please combine these two into one.

Reply:
Also, this will always return a `JSONObject`, event when one with the desired `key` is not found.

Reply:
Assign `jsonObject = jsonArray.optJSONObject(i);` within the `if` block instead.